base/docker/scripts/connection_pool.py

```diff
@@ -0,0 +1,95 @@
+import threading
```

Comment:
This can benefit SDK, should be moved there. thoughts?

Reply:
This specific version always uses `CommonLibrary.create_connector()`, which is what we want for workflows. I agree with moving it to SDK, but you want more control over connector creation.

Reply:
OK, we shall make this refactoring outside of this PR. https://github.com/aperture-data/aperturedb-python/issues/598

Created an issue to flesh it out.